docker/Dockerfile

```diff
@@ -0,0 +1,291 @@
+# This Dockerfile can be built from scratch with e.g the command
+# docker build . -t my_numba_dpex_build
+# The ARGs in the header of this file can be changed as needed (either by
+# editing the file or with the command line).
+# The build can be fairly long (up to one hour on low-end machines.)
+# To start a container, use e.g
+# sudo docker run --rm --name my_container -it -v /my/host/volume/:/mounted/volume --device=/dev/dri my_numba_dpex_build
+# The parameter --device=/dev/dri enables gpu passthrough (as long as the user 
+# has permissions, e.g by being a member of the video group)
+
+
+# TODO: make a custom build of the base image to test with other base OS
+# (e.g ubuntu2204)
+# ???: actually not sure if it's needed or interesting to use the sycl nightly
+# binaries at all. The nightly also ship an up-to-date installation of the
+# intel runtime, including drivers and opencl and level_zero runtimes.
+
+ARG BASE=ghcr.io/intel/llvm/sycl_ubuntu2004_nightly
+ARG TMPDIR=/tmp
```

Comment:
Thank you for rewording, a lot of comments are still personal notes I wrote down in the process rather than appropriate documentation, and the file is not in a good shape overall.

About the use of sycl_ubuntu2004_nightly, to begin with I chose the image because it ships up-to-date drivers and runtime for opencl and level_zero, and for now it strips away the complexity of those steps from the current PR, which can be more focused on getting the numba-dpex installation right. The dockerfiles for this base are included in the llvm repo and looks sensible which made me think this base is reliable + it benefits from daily CI. It's of course possible to start from any other base (preferably debian-like) image and script the driver installation (it's will even be necessary to test compatibility with other OSes), at the cost of some more complexity.

If the use of the nightly sycl binaries turns out to be unneeded or detrimental, `ghcr.io/intel/llvm/ubuntu2004_intel_drivers` can be used instead, it's the same base without sycl binaries.

You're right about pinning the version of the base image, will add.